src/python/pants/backend/python/targets/python_distribution.py

```diff
@@ -61,3 +61,7 @@ def __init__(self,
         PythonIdentity.parse_requirement(req)
       except ValueError as e:
         raise TargetDefinitionException(self, str(e))
+
+  @property
+  def has_native_sources(self):
+    return any(src.endswith(('.c', '.cpp', '.cc')) for src in self.sources_relative_to_target_base())
```

Comment:
This can be accomplished with `self.has_sources(extension=('.c', '.cpp', '.cc'))`.